gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -62,19 +62,8 @@ case class TransformPreOverrides() extends Rule[SparkPlan] {
   : SparkPlan = {
     val hashExpression = new Murmur3Hash(exprs)
     hashExpression.withNewChildren(exprs)
-    val project = child match {
-      case exec: ProjectExec =>
-        // merge the project node
```

Comment:
If without this step, it will have 2 continuous projectNode, why it can cause fallback?

Reply:
Let's suppose the plan is `project_2 -> project_1 -> child_of_project_1`, actually what happens here is to connect `project_2` with `child_of_project_1`, which causes binding error during validation. Also, because Velox projection has lazy mechanism, so I guess two projects here will not bring much performance regression, but I'll make some tests to verify.

Reply:
project_2 is project_1 + hash_partition_key. Now it is just used for velox, but may be used for other backend such as clickhouse later. And I think keep 2 projectionNode is not a good way, can we correct the binding?

Reply:
With previous logic, `project_2` only selects `project_1`'s output and computes hash, but `project_1`'s computing is missing. That's why there is the binding issue.